### PR TITLE
Refactor Shape.Rectangle.RotateToIsometry

### PR DESCRIPTION
**Pull request by reaperrr**
Also fixes TS wall and gate shapes.

### CODE REVIEW

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 1]
```diff
@@ -600,6 +600,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
...
 					if (node.Key == "SoundFile" && parent.Key.StartsWith("AmbientSound", StringComparison.Ordinal))
 						RenameNodeKey(node, "SoundFiles");
+				
+				// Refactor Rectangle shape RotateToIsometry bool into WAngle HorizontalRotation
+				if (engineVersion < 20170505)
+				{
+					if (node.Key.StartsWith("RotateToIsometry", StringComparison.Ordinal))
+					{
+						var value = FieldLoader.GetValue<string>("RotateToIsometry", node.Value.Value);
+						if (value == "true")
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: You'll need use `value.ToLower()` and compare against `true` and `yes`.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 2]
```diff
@@ -601,6 +601,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					if (node.Key == "SoundFile" && parent.Key.StartsWith("AmbientSound", StringComparison.Ordinal))
 						RenameNodeKey(node, "SoundFiles");
 
+				// Refactor Rectangle shape RotateToIsometry bool into WAngle HorizontalRotation
+				if (engineVersion < 20170505)
+				{
+					if (node.Key.StartsWith("RotateToIsometry", StringComparison.Ordinal))
+					{
+						var value = FieldLoader.GetValue<string>("RotateToIsometry", node.Value.Value).ToLower();
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: `GetValue<bool>`, and then you avoid duplicating FieldLoader's boolean parsing.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 3]
```diff
@@ -601,6 +601,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
...
+				// Refactor Rectangle shape RotateToIsometry bool into WAngle HorizontalRotation
+				if (engineVersion < 20170505)
+				{
+					if (node.Key.StartsWith("RotateToIsometry", StringComparison.Ordinal))
+					{
+						var value = FieldLoader.GetValue<bool>("RotateToIsometry", node.Value.Value);
+						if (value == true)
+							node.Value.Value = "128";
+						else
+							node.Value.Value = "0";
```
Comment (aydjay) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: You may want to use a ternary here for legibility.

'node.Value.Value = value ? "128" : "0";'

File: OpenRA.Mods.Common/HitShapes/Rectangle.cs [hunk 4]
```diff
@@ -33,10 +33,9 @@ public class RectangleShape : IHitShape
 		[Desc("Defines the bottom offset relative to the actor's target point.")]
 		public readonly int VerticalBottomOffset = 0;
 
-		// This is just a temporary work-around until we have a customizable PolygonShape
-		[Desc("Rotates shape by 90 degree relative to actor facing. Mostly required for buildings on isometric terrain.",
+		[Desc("Rotates shape by an angle relative to actor facing. Mostly required for buildings on isometric terrain.",
 			"Mobile actors do NOT need this!")]
-		public readonly bool RotateToIsometry = false;
+		public readonly WAngle HorizontalRotation = WAngle.Zero;
```
Comment (pchote) [inline OpenRA.Mods.Common/HitShapes/Rectangle.cs]: Please call this `LocalYaw` for consistency with our other yaw angle offsets.

**Review by pchote — APPROVED** [review]

**Comment by reaperrr** [comment]
Updated, greatly condensed `var orientation` in the hitshape by using WRot.FromYaw.
Still works as before according to my testing, but requesting @pchote to throw this a last glance.

**Comment by pchote** [comment]
My :+1: still holds.

**Comment by reaperrr** [comment]
Depends on #13253.

**Comment by reaperrr** [comment]
Rebased and updated to account for #13253.

**Comment by pchote** [comment]
My :+1: *still* still holds.

I noticed a couple of things that could be fixed here as a scope creep, or left to a future PR:
* The blocking height of walls and gates are different.
* Component towers and laser fence posts are parts of walls, so should have the same hitshape.

**Review by chrisforbes — APPROVED** [review]

**Pull request merged**